include/ProxySQL_Cluster.hpp

```diff
@@ -117,20 +117,87 @@ class ProxySQL_Node_Entry {
 	uint64_t global_checksum;
 };
 
+struct p_cluster_nodes_counter {
+	enum metric {
+		__size
+	};
+};
+
+struct p_cluster_nodes_gauge {
+	enum metric {
+		__size
+	};
+};
+
+struct p_cluster_nodes_dyn_counter {
+	enum metric {
+		proxysql_servers_checksums_version_total,
+		proxysql_servers_metrics_uptime_s,
+		proxysql_servers_metrics_queries,
+		proxysql_servers_metrics_client_conns_created,
+		__size
+	};
+};
+
+struct p_cluster_nodes_dyn_gauge {
+	enum metric {
+		proxysql_servers_checksums_epoch,
+		proxysql_servers_checksums_updated_at,
+		proxysql_servers_checksums_changed_at,
+		proxysql_servers_checksums_diff_check,
+		proxysql_servers_metrics_weight,
+		proxysql_servers_metrics_response_time_ms,
+		proxysql_servers_metrics_last_check_ms,
+		proxysql_servers_metrics_client_conns_connected,
+		__size
+	};
+};
+
+struct cluster_nodes_metrics_map_idx {
+	enum index {
+		counters = 0,
+		gauges,
+		dyn_counters,
+		dyn_gauges
+	};
+};
+
 class ProxySQL_Cluster_Nodes {
 	private:
 	pthread_mutex_t mutex;
 	std::unordered_map<uint64_t, ProxySQL_Node_Entry *> umap_proxy_nodes;
 	void set_all_inactive();
 	void remove_inactives();
 	uint64_t generate_hash(char *_hostname, uint16_t _port);
+	struct {
+		std::array<prometheus::Family<prometheus::Counter>*, p_cluster_nodes_dyn_counter::__size> p_dyn_counter_array {};
+		std::array<prometheus::Family<prometheus::Gauge>*, p_cluster_nodes_dyn_gauge::__size> p_dyn_gauge_array {};
+
+		// proxysql_servers_checksum
+		std::map<std::string, prometheus::Counter*> p_proxysql_servers_checksum_version {};
+		std::map<std::string, prometheus::Gauge*> p_proxysql_servers_checksums_epoch {};
```

Comment:
yeah... but if we are going to make a convention we should make it in a later PR. I'm saying this just because the codebase has already lots of definitions like these ones. So, in a later PR we can propose shorten all of them, even if for these particular case I don't see much trouble, I like more the shortening of names when they need to be mentioned multiple times, but I like to keep the definitions easy to read.